uhubctl.c

```diff
@@ -1051,13 +1092,427 @@ static int usb_find_hubs(void)
+    if (libusb_get_device_descriptor(dev, &desc) == 0) {
+        char vendor_id[8], product_id[8];
+        snprintf(vendor_id, sizeof(vendor_id), "0x%04x", desc.idVendor);
+        snprintf(product_id, sizeof(product_id), "0x%04x", desc.idProduct);
+        cJSON_AddStringToObject(port_json, "vendor_id", vendor_id);
+        cJSON_AddStringToObject(port_json, "product_id", product_id);
+        
+        cJSON_AddNumberToObject(port_json, "device_class", desc.bDeviceClass);
+        
+        const char* class_name = get_primary_device_class_name(dev, &desc);
+        cJSON_AddStringToObject(port_json, "device_class_name", class_name);
+
+        // Add interface information
+        struct libusb_config_descriptor *config;
+        if (libusb_get_config_descriptor(dev, 0, &config) == 0) {
+            cJSON* interfaces = cJSON_CreateArray();
+            for (int i = 0; i < config->bNumInterfaces; i++) {
+                const struct libusb_interface *interface = &config->interface[i];
+                for (int j = 0; j < interface->num_altsetting; j++) {
+                    const struct libusb_interface_descriptor *altsetting = &interface->altsetting[j];
+                    cJSON* intf = cJSON_CreateObject();
+                    cJSON_AddNumberToObject(intf, "interface_number", i);
+                    cJSON_AddNumberToObject(intf, "alt_setting", j);
+                    cJSON_AddNumberToObject(intf, "interface_class", altsetting->bInterfaceClass);
+                    cJSON_AddStringToObject(intf, "interface_class_name", get_class_name(altsetting->bInterfaceClass));
+                    cJSON_AddItemToArray(interfaces, intf);
+                }
+            }
+            cJSON_AddItemToObject(port_json, "interfaces", interfaces);
+            libusb_free_config_descriptor(config);
+        }
+
+        // Add serial number if available
+        if (desc.iSerialNumber) {
+            struct libusb_device_handle *devh;
+            if (libusb_open(dev, &devh) == 0) {
+                unsigned char serial[256];
+                int ret = libusb_get_string_descriptor_ascii(devh, desc.iSerialNumber, serial, sizeof(serial));
+                if (ret > 0) {
+                    cJSON_AddStringToObject(port_json, "serial_number", (char*)serial);
+                }
+                libusb_close(devh);
+            }
+        }
+
+        // Add USB version
+        char usb_version[8];
+        snprintf(usb_version, sizeof(usb_version), "%x.%02x", desc.bcdUSB >> 8, desc.bcdUSB & 0xFF);
+        cJSON_AddStringToObject(port_json, "usb_version", usb_version);
+
+        // Add device version
+        char device_version[8];
+        snprintf(device_version, sizeof(device_version), "%x.%02x", desc.bcdDevice >> 8, desc.bcdDevice & 0xFF);
+        cJSON_AddStringToObject(port_json, "device_version", device_version);
+
+        // Add number of configurations
+        cJSON_AddNumberToObject(port_json, "num_configurations", desc.bNumConfigurations);
```

Comment:
Emitting this makes output quite verbose. Do we really need to know number of configurations for every device?